### PR TITLE
Getting those MSVC Build Scripts Rolling

### PR DESCRIPTION
**Pull request by valarauca**
See: https://github.com/cogciprocate/ocl/issues/28

This allows the crate to compile successfully on Windows10 x64 MSVC (should handle x86 too). I have an RX480, so I'm unsure how _other_ AMD cards will handle their OpenCL SDK. But seeing as AMD only offer 1 OpenCL SDK download I think they should be consistent. 

This weekend I should be getting my Nvidia Card back (was attempting to sell it on ebay). I'll throw its OpenCL SDK in also.

Should we add feature flag for which backend is being used? As it is likely possible the developer has an Intel, and Nvidia/AMD SDK installed at the same time.

:.:.:

Should the buffer_copy tests take >60 seconds?

### CODE REVIEW

**Comment by c0gent** [comment]
Very awesome :)

I'll test on some windows machines tomorrow, I have some AMD, Nvidia, and Intel combinations avail.

The buffer_copy tests should not take that long so you may want to open a separate issue for that and we can get to the bottom of it.

**Comment by valarauca** [comment]
I wrote the patch at about 11:30pm, and started testing it around midnight with the full `ocl` library. So I wasn't fully able to diagnosis why the tests weren't finishing.

File: README.md [hunk 1]
```diff
@@ -25,3 +25,14 @@ at your option.
 Unless you explicitly state otherwise, any contribution intentionally submitted
 for inclusion in the work by you, as defined in the Apache-2.0 license, shall be dual licensed as above, without any
 additional terms or conditions.
+
+##### BROKEN ON WINDOWS MSVC
```
Comment (c0gent) [inline README.md:29]: This isn't exactly true, everything works fine on MSVC as long as you put your OpenCL.lib in the right spots (I need to check the details, it's been a long time since I've played with this on Windows). Perhaps something like 'MSVC Caveats' might be a better title.

File: README.md [hunk 2]
```diff
@@ -25,3 +25,14 @@ at your option.
...
+
+Download your Vendors OpenCL SDK:
```
Comment (c0gent) [inline README.md:31]: Users do not need to install the SDK in the majority of cases. The appropriate OpenCL drivers are often included with standard drivers. Again I'll need to double check the details on this and I'll help you make tweaks to this description.

**Review by c0gent — COMMENTED** [review]
Haven't used this new review system so giving it a try.

**Comment by c0gent** [comment]
Haven't had a chance to actually test your build script on Windows yet. I'll get to it asap.

> Should we add feature flag for which backend is being used? As it is likely possible the developer has an Intel, and Nvidia/AMD SDK installed at the same time.

It's impractical and unnecessary to do anything platform specific, simply due to the fact that, as you mentioned, users may have multiple installed platforms but only using one or a subset of them at any given time. As long as we search all known possible install locations at compile time, we're good (your documentation here is exactly right, users will have to add/report new locations).

I've added a couple of notes on your edits to `README.md`.

**Comment by c0gent** [comment]
Builds great for me on Windows-GNU-[Nvidia/AMD/Intel] and Windows-MSVC-[Nvidia/AMD/Intel] on two different machines (using rustup installs).

I would need to test it on a clean machine though because I can't remember exactly what paths I've set up that are probably making the linkers job easy. I messed around with running it in a VM but I can't get OpenCL drivers to install at all and I just don't have time to mess with it unfortunately.

So basically we need more testing but... it works for me (famous last words).

**Comment by c0gent** [comment]
@valarauca 

I'm inclined to merge it and see what reports we get from users building from the repo. As there probably aren't very many this would take a while to get feedback.  Let me know what you think.

I'll probably also have more time to set up a good test machine this weekend or next week I'm just slammed right now.

**Comment by valarauca** [comment]
Cool I had no issues building the cl-sys library, just some of the tests in the higher ocl library started failing, or just running forever. That was my hesitation.

**Comment by c0gent** [comment]
Yes. Let's troubleshoot the and solve that problem (ocl running forever or failing) first. Let's make sure it's not an underlying problem of some sort with the library.  

When you have time post some details about it and we'll get to the bottom of it.

**Comment by Kerollmops** [comment]
Ooops [That's not the expected path I think](https://github.com/cogciprocate/ocl/blob/master/Cargo.toml#L27) ?

**Comment by c0gent** [comment]
Yeah, sorry if I pushed that to the repo (not the first time).  Feel free to PR that and I'll merge it. I'm not at home otherwise I would.

**Comment by c0gent** [comment]
@valarauca @Kerollmops 

Going to merge this to master unless there are any objections. If there are problems on other platforms we'll just have to wait for issues to be filed.

**Comment by c0gent** [comment]
@valarauca 

Please file an issue on the ocl repo about the running forever / failing problem if you're still having it.

**Comment by valarauca** [comment]
Can do. I'll have to re-check.

**Comment by c0gent** [comment]
Going to merge then make some clarifications to your comments.

Keep me up to date on what you find out about that issue.

If users have issues I hope you don't mind maintaining the build script if I can't. I'm stretched really thin right now. 

Thanks a lot for your work.
